src/webapp/pages/bulk-apply/steps/useSummaryStep.tsx

```diff
@@ -0,0 +1,80 @@
+import { useCallback, useEffect, useState } from "react";
+import { useAppContext } from "../../../contexts/app-context";
+import { SharingUpdate } from "../../../../domain/entities/SharingUpdate";
+import { ImportResult } from "../../../../domain/entities/ImportResult";
+import { SharingSummary } from "../../../../domain/entities/SharingSummary";
+import { NamedRef } from "../../../../domain/entities/Ref";
+
+type GlobalMessage = {
+    text: string;
+    type: "success" | "error";
+};
+
+export function useSummaryStep(builder: SharingUpdate) {
+    const { compositionRoot } = useAppContext();
+
+    const [importResult, setImportResult] = useState<ImportResult>();
+    const [openDialog, setDialogOpen] = useState<boolean>(false);
+    const [sharingSummary, setSharingSummary] = useState<SharingSummary>();
+    const [loading, setLoading] = useState<boolean>(false);
+    const [globalMessage, setGlobalMessage] = useState<GlobalMessage | undefined>(undefined);
+    const [excludedMetadata, setExcludedMetadata] = useState<NamedRef[]>([]);
+
+    useEffect(() => {
+        compositionRoot.metadata.getExcludedMetadata(builder.excludedDependencies).run(
```

Comment:
Here you have to show excluded dependencies info if it's a dependency from selected data in step 1 (these dependencies are listed in the step 2). You can do this in the use case GetExcludedDependencies to get the list of dependencies of the selection in step 1 and filter them (see ListDependenciesUseCase)